### DISCUSSION TITLE
Support for a CSS include file

### DISCUSSION THREAD

**Discussion by ianpmurphy** [grafana/grafana#64894] [Dashboards, Panels & Visualizations]
Its highly desirable to be able to format modules like dynamictext or rss/atom from a single point. If you have multiple text panels on multiple dashboards its messy to have to modify them one by one to include a small change to, say, text colour or size and to make them all the same. It should be done via CSS.

While it is possible to include a css file using the boom theme, it requires you to have a panel somewhere on the screen. The use of a dedicated panel has a problem where, if the panel doesn't fit onscreen then the css doesn't kick in.

**The Request**
A dashboard should have an option to include a url to a css file which is added to the page header. No need for anything more.

**Comment by FidelAhalCjW** [comment]
Yeah, I totally get what you're saying. It can be a real hassle to have to go in and manually tweak every single module just to make a small change like adjusting the text color or size. It would be so much easier if we could just set those parameters from a central location via CSS.

I see what you mean about the Boom theme - it's cool that you can include a CSS file, but it's kind of a pain that you have to have a panel on the screen for it to work. And if the panel is too big, you're out of luck.

So, if I'm understanding you correctly, you're suggesting that there should be a way to include a URL to a CSS file for the entire dashboard, without having to mess around with panels or anything like that? That sounds like a great idea to me! It would definitely make things a lot easier and more efficient.

**Comment by ianpmurphy** [comment]
yep, exactly that. 

I think a distinct option per dashboard would be the ideal solution as there may be many dashboards in an instalation and they may not have anything to do with each other.

I would imagine that it doesn't require much work for the development team either as it doesn't change the functionality of Grafana itself.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82849. Feel free to continue the discussion around this there. Thank you!
